src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -14,43 +14,142 @@ internal static class CompletionHelpers
+        ///     <item><description>Replaces "\0" (null) with "`0".</description></item>
+        ///     <item><description>Replaces "\a" (bell) with "`a".</description></item>
+        ///     <item><description>Replaces "\b" (backspace) with "`b".</description></item>
+        ///     <item><description>Replaces "\u001b" (escape character) with "`e".</description></item>
+        ///     <item><description>Replaces "\f" (form feed) with "`f".</description></item>
+        ///     <item><description>Replaces "\v" (vertical tab) with "`v".</description></item>
+        /// </list>
+        /// </remarks>
+        internal static string NormalizeToExpandableString(string value)
+            => value
+                .Replace("\r", "`r")
+                .Replace("\n", "`n")
+                .Replace("\t", "`t")
+                .Replace("\0", "`0")
+                .Replace("\a", "`a")
+                .Replace("\b", "`b")
+                .Replace("\u001b", "`e")
+                .Replace("\f", "`f")
+                .Replace("\v", "`v");
+
+        /// <summary>
+        /// Defines a strategy for determining if a value matches a word or pattern.
+        /// </summary>
+        /// <param name="value">The input string to check for a match.</param>
+        /// <param name="wordToComplete">The word or pattern to match against.</param>
+        /// <returns>
+        /// <c>true</c> if the value matches the specified word or pattern; otherwise, <c>false</c>.
+        /// </returns>
+        internal delegate bool MatchStrategy(string value, string wordToComplete);
+
+        /// <summary>
+        /// Determines if the given value matches the specified word using a literal, case-insensitive prefix match.
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if the value starts with the word (case-insensitively); otherwise, <c>false</c>.
+        /// </returns>
+        internal static readonly MatchStrategy LiteralMatch = (value, wordToComplete)
+            => value.StartsWith(wordToComplete, StringComparison.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Determines if the given value matches the specified word using wildcard pattern matching.
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if the value matches the word as a wildcard pattern; otherwise, <c>false</c>.
+        /// </returns>
+        /// <remarks>
+        /// Wildcard pattern matching allows for flexible matching, where wilcards can represent 
+        /// multiple characters in the input. This strategy is case-insensitive.
+        /// </remarks>
+        internal static readonly MatchStrategy WildcardPatternMatch = (value, wordToComplete)
+            => WildcardPattern
+                .Get(wordToComplete + "*", WildcardOptions.IgnoreCase)
+                .IsMatch(value);
+
+        /// <summary>
+        /// Determines if the given value matches the specified word using either a literal or wildcard escape strategy.
```

Comment:
It's hard to understand the idea how it works. Since remarks section already have details I think we should rephrase the summary:
```suggestion
        /// Determines if the given value matches the specified word considering wildcard characters literally.
```